jdee-flycheck.el

```diff
@@ -212,6 +212,12 @@ file and buffer with the contents of the current buffer and compiles that one."
       (unless jdee-compile-option-directory
         (set (make-local-variable 'jdee-compile-option-directory)
              (expand-file-name "classes" dir)))
+      ;; Use jdee-compile-option-directory dir from orig-buffer if it is set
+      ;; as a local variable. When compiling a test file this dir is normally
+      ;; test-classes, not classes. Runtime classes and test classes have
+      ;; different jdee-compile-option-directory in the same project.
+      (let ((v (buffer-local-value 'jdee-compile-option-directory orig-buffer)))
+        (if v (set 'jdee-compile-option-directory v)))
```

Comment:
But where ```jdee-compile-option-directory``` will be set to *test-classes* value? You would need to recognize if you are in test sources or _main_ sources.

Reply:
Sorry for such a long reply.

Reply:
In prj.el it is possible to set the jdee-compile-option-directory based in the full path filename, in a local version of the variable. In my case I do it like this:
```elisp
(setq
   jdee-compile-option-directory
   (cond
    ((string-match
      ".*\\(/\\(src/main/java\\)/.*\\)" default-directory)
     (replace-match "/target/classes/" t nil default-directory 1))
    ((string-match
      ".*\\(/\\(src/test/java\\)/.*\\)" default-directory)
     (replace-match "/target/test-classes/" t nil default-directory 1))

    (t (error "%s" (concat "Destiny of '" (buffer-file-name)
			   "' compiled file is unknown")))) )
```
With the change jdee-flycheck can give one a chance to set the compiled file location. I hope this patch can flourish a better idea!